packages/guides/src/Compiler/Passes/MoveAnchorPass.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\Compiler\Passes;
+
+use phpDocumentor\Guides\Compiler\CompilerPass;
+use phpDocumentor\Guides\Nodes\AnchorNode;
+use phpDocumentor\Guides\Nodes\CompoundNode;
+use phpDocumentor\Guides\Nodes\DocumentNode;
+
+class MoveAnchorPass implements CompilerPass
```

Comment:
Shouldn't this be a node transformer instead? That already has all code to traverse the tree of nodes.

The way I differentiate between them is:
* Node transformers operate within a single document (and therefor can even be cached)
* Compiler passes operate in the project, managing cross document logic (like toctrees).

Reply:
I think the current node tranformer cannot handle this, as I need to move elements. But I could be wrong... 
I have to make sure that I move elements at the right level, and in the node transformer, I don't know that?

Reply:
Hmm, maybe I have a wrong idea about the AST that we are dealing with here. But if it looks like this:

```
COMPOUND
    ...
ANCHOR
ANCHOR
COMPOUND
    ...
```

I think a transformer that supports compound and anchor nodes should work. It'll then store the anchors in a property when it discovers them, and push them in the compound node when it reaches one (clearing the property).

Reply:
I will check if it is possible to do this in a transformer, I think the main issue I will have is that the transformer will first handle the child nodes an then go to the next node. And its state is shared between the nodes. 

ANCHOR
COMPOUND
    ANCHOR
    ANCHOR
    COMPOUND
    ANCHOR
COMPOUND

Should be come

COMPOUND
    ANCHOR
    COMPOUND
    ANCHOR
    ANCHOR
COMPOUND
    ANCHOR